src/lib.rs

```diff
@@ -1912,11 +1914,7 @@ pub unsafe trait FromBytes: FromZeros {
     where
         Self: KnownLayout + Immutable,
     {
-        util::assert_dst_is_not_zst::<Self>();
```

Comment:
These assertions should be retained here. It really helps with the error messages to have const asserts as high up on the call stack as possible.

Reply:
Done. In fact, this was incorrect anyway since it'd error in the case where an explicit count is provided.